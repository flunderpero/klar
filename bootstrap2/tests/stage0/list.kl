struct Person:
    name str
    age i32
end

fn get_person(list List, index i32) Person:
    return list_get(list, index)
end

fn main():
    let list = list_new()
    print(f"len: {list.len}, cap: {list.cap}")
    list_push(list, Person("Alice", 20))
    print(f"len: {list.len}, cap: {list.cap}")
    list_push(list, Person("Bob", 21))
    print(f"len: {list.len}, cap: {list.cap}")
    list_push(list, Person("Peter", 22))
    print(f"len: {list.len}, cap: {list.cap}")
    list_push(list, Person("Paul", 23))
    print(f"len: {list.len}, cap: {list.cap}")
    list_push(list, Person("Mary", 24))
    print(f"len: {list.len}, cap: {list.cap}")
    mut i = 0
    while i < list.len:
        let p = get_person(list, i)
        print(f"{p.name} is {p.age} years old.")
        i = i + 1
    end
end
