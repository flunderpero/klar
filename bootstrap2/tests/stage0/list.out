len: 0, cap: 1
len: 1, cap: 1
len: 2, cap: 2
len: 3, cap: 4
len: 4, cap: 4
len: 5, cap: 8
Alice is 20 years old.
Bob is 21 years old.
Peter is 22 years old.
Paul is 23 years old.
Mary is 24 years old.
